src/odemis/gui/comp/positions.py

```diff
@@ -0,0 +1,254 @@
+            flag=wx.ALL | wx.ALIGN_CENTER_VERTICAL,
+            border=5,
+        )
+        lbl_ctrl = wx.StaticText(self._panel, -1, "Y")
+        self.gb_sizer.Add(
+            lbl_ctrl,
+            (self.num_rows, 2),
+            flag=wx.ALL | wx.ALIGN_CENTER_VERTICAL,
+            border=5,
+        )
+        self.num_rows += 1
+
+        trans_cor = stage_md[model.MD_POS_TRANS_COR]
+        pos_cor = stage_md[model.MD_POS_COR]
+
+        conf = {"unit": "m", "accuracy": 3}
+
+        pos: dict = {
+            "Rotation": pos_cor,
+            "Translation": trans_cor,
+        }
+
+        for name, val_cor in pos.items():
+            lbl_ctrl = wx.StaticText(self._panel, -1, name)
+            self.gb_sizer.Add(
+                lbl_ctrl,
+                (self.num_rows, 0),
+                flag=wx.ALL | wx.ALIGN_CENTER_VERTICAL,
+                border=5,
+            )
+            x_val, y_val = val_cor
+
+            x_value_ctrl = UnitFloatCtrl(
+                self._panel, value=x_val, style=wx.NO_BORDER, **conf
+            )
+            y_value_ctrl = UnitFloatCtrl(
+                self._panel, value=y_val, style=wx.NO_BORDER, **conf
+            )
+            self.gb_sizer.Add(
+                x_value_ctrl,
+                (self.num_rows, 1),
+                flag=wx.ALL | wx.EXPAND | wx.ALIGN_CENTER_VERTICAL,
+                border=5,
+            )
+            self.gb_sizer.Add(
+                y_value_ctrl,
+                (self.num_rows, 2),
+                flag=wx.ALL | wx.EXPAND | wx.ALIGN_CENTER_VERTICAL,
+                border=5,
+            )
+
+            self.ctrl_dict[name] = {"x": x_value_ctrl, "y": y_value_ctrl}
+
+            x_value_ctrl.SetForegroundColour(gui.FG_COLOUR_EDIT)
+            x_value_ctrl.SetBackgroundColour(gui.BG_COLOUR_MAIN)
+            y_value_ctrl.SetForegroundColour(gui.FG_COLOUR_EDIT)
+            y_value_ctrl.SetBackgroundColour(gui.BG_COLOUR_MAIN)
+            self.num_rows += 1
+
+        # add update button
+        self.update_button = wx.Button(self._panel, -1, "Update")
+        self.update_button.Bind(wx.EVT_BUTTON, self._on_update)
+        self.gb_sizer.Add(
+            self.update_button,
+            (self.num_rows, 0),
+            flag=wx.ALL | wx.ALIGN_CENTER_VERTICAL,
+            border=5,
+        )
+        self.update_button = wx.Button(self._panel, -1, "Cancel")
+        self.update_button.Bind(wx.EVT_BUTTON, self._on_cancel)
+        self.gb_sizer.Add(
+            self.update_button,
+            (self.num_rows, 1),
+            flag=wx.ALL | wx.ALIGN_CENTER_VERTICAL,
+            border=5,
+        )
+
+        # Fit sizer
+        self.main_sizer.AddSpacer(5)
+        self.Fit()
+        self.Layout()
+
+    def _on_cancel(self, event):
+        logging.debug("Cancel button clicked")
+        event.Skip()
+        self.Close()
+
+    def _on_update(self, event):
+        logging.debug("Update button clicked")
+        event.Skip()
```

Comment:
Most of the time, you don't need to skip wx events. Is it necessary here? If so, please add a comment explaining why.
Same for `_on_cancel()`